SyslogNet.Client/Transport/SyslogTcpSender.cs

```diff
@@ -118,4 +129,14 @@ public void Dispose()
 			}
 		}
 	}
+
+	static class MemoryStreamExtensions
+	{
+		private static readonly byte[] WriteByteAsyncBuffer = new byte[1];
+		public static Task WriteByteAsync(this MemoryStream stream, byte value)
+		{
+			WriteByteAsyncBuffer[0] = value;

```

Comment:
This isn't threadsafe (two calls can write to the static array and then both write whatever the last one wrote).

Making this a ThreadLocal<byte[]> that initializes to a new byte[1] would be a simple way to make this thread safe while keeping the number of allocations very low. I don't think this is a case where a more complex pooling mechanism would bring benefit as we would have at max one byte[1] for each thread.

Note that the synchronous Stream.WriteByte method does do a new allocation for each call so using a ThreadLocal would be a slight improvement over the allocations currently made for the sync version.
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/IO/Stream.cs#L757